docs/eloquent-models/schema-builder.txt

```diff
@@ -0,0 +1,388 @@
+the ``Schema`` facade in your migration file. You can use this to
+perform migration logic conditionally.
+
+The following example migration creates a ``stars`` collection if a collection
+named ``telescopes`` exists:
+
+.. literalinclude:: /includes/schema-builder/stars_migration.php
+   :language: php
+   :dedent:
+   :start-after: begin conditional create
+   :end-before: end conditional create
+
+.. _laravel-eloquent-indexes:
+
+Manage Indexes
+--------------
+
+MongoDB indexes are data structures that improve query efficiency by reducing
+the number of documents needed to retrieve query results. Indexes such as
+geospatial indexes extend how you can query the data.
+
+To improve query performance by using an index, make sure the index covers
+the query.  To learn more about indexes and query optimization, see the
+following MongoDB server manual entries:
+
+- :manual:`Indexes </indexes>`
+- :manual:`Query Optimization </core/query-optimization/>`
+
+The following sections show how you can use the schema builder to create and
+drop various types of indexes on a collection.
+
+Create an Index
+~~~~~~~~~~~~~~~
+
+To create indexes, call the ``create()`` method on the ``Schema`` facade
+in your migration file. Pass it the collection name and a callback
+method with a ``MongoDB\Laravel\Schema\Blueprint`` parameter. Specify the
+index creation details on the ``Blueprint`` instance.
+
+The following example migration creates indexes on the following collection
+fields:
+
+- Single field index on ``mission_type``
+- Compound index on ``launch_location`` and ``launch_date``, specifying a descending sort order on ``launch_date``
+- Unique index on the ``mission_id`` field, specifying the index name "unique_mission_id_idx"
+
+Click the :guilabel:`VIEW OUTPUT` button to see the indexes created by running
+the migration, including the default index on the ``_id`` field:
+
+.. io-code-block::
+
+   .. input:: /includes/schema-builder/flights_migration.php
+      :language: php
+      :dedent:
+      :start-after: begin create index
+      :end-before: end create index
+
+   .. output::
+      :language: json
+      :visible: false
+
+      [
+        { v: 2, key: { _id: 1 }, name: '_id_' },
+        { v: 2, key: { mission_type: 1 }, name: 'mission_type_1' },
+        {
+          v: 2,
+          key: { launch_location: 1, launch_date: -1 },
+          name: 'launch_location_1_launch_date_-1'
+        },
+        {
+          v: 2,
+          key: { mission_id: 1 },
+          name: 'unique_mission_id_idx',
+          unique: true
+        }
+      ]
+
+Specify Index Options
+~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB index options determine how the indexes are used and stored.
+You can specify index options when calling an index creation method such
+as ``index()`` on a ``Blueprint`` instance.
+
+The following migration code shows how to add a collation to an index as an
+index option. Click the :guilabel:`VIEW OUTPUT` button to see the indexes
```

Comment:
Suggestion: move this to its own line so it doesn't seem like it's supposed to be explaining what the example does further
```suggestion
index option. 

Click the :guilabel:`VIEW OUTPUT` button to see the indexes
```

Reply:
Thanks for the suggestion. I tried a few different revisions before deciding on this. I think we usually introduce the example in the paragraph directly before the example, so this seemed like the best option. I used this pattern in another related PR: https://github.com/mongodb/laravel-mongodb/pull/2747

Do you feel strongly about changing this? Let me know what you think.

Reply:
Based on our Slack conversation, I'll leave it for now and make updates if any users complain.